### PR TITLE
Issue #2826 Traitor Context menu correction (no traitoring to teamless)

### LINKED ISSUE #2826: 48.0 Can traitor units to a player with no assigned team
### Description
If a player joins a game after it starts, they have no assigned team by default. However, pre-existing players can traitor units to said teamless player(s). If they do so, said units will never get a chance to move or otherwise act, as the player without an assigned team has no place in the initiative order. 

Uncertain exactly how to fix this (prevent players without an assigned team from appearing in the Traitor Units target list?), but it can cause some confusing situations. 

Autosaves from a quickly thrown together demonstration game that was used to verify that the error reproduced:
Firing phase prior to transfer - [autosave_2021-05-03_23-26-28.sav.gz](https://github.com/MegaMek/megamek/files/6418923/autosave_2021-05-03_23-26-28.sav.gz)
Initiative phase post-transfer - [autosave_2021-05-03_23-26-32.sav.gz](https://github.com/MegaMek/megamek/files/6418926/autosave_2021-05-03_23-26-32.sav.gz)

Unit was transferred to "Interloper" player (as can be seen in the round report of the second autosave). Note that said player does not appear in the initiative order of the initiative phase in the second autosave. Other players were for setup purposes only and can be replaced with bots for testing purposes. You can also test the /joinTeam command (in my test this allowed the Interloper units to move once the player had been assigned a team).

### PR DESCRIPTION
**Pull request by SJuliez**
Changes the context menu to only allow traitoring units to players assigned to a team, same as the /traitor command. Also shows the menu item only during the movement phase as it works only there anyway. 
I first thought the issue was with the /traitor command, therefore some beauty changes there and in the joinTeamCommand. 

Resolves #2826

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/MapMenu.java [hunk 1]
```diff
@@ -250,22 +216,20 @@ private boolean createMenu() {
...
-                        ex.printStackTrace();
+            JMenuItem item = new JMenuItem(Messages.getString("MovementDisplay.Traitor"));
+            item.setActionCommand(MovementDisplay.MoveCommand.MOVE_TRAITOR.getCmd());
+            item.addActionListener(e -> {
+                try {
+                    if (currentPanel instanceof MovementDisplay) {
+                        ((MovementDisplay) currentPanel).actionPerformed(e);
                     }
+                } catch (Exception ex) {
+                    ex.printStackTrace();
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/MapMenu.java:227]:
```suggestion
                    MegaMek.getLogger().error(ex);
```

File: megamek/src/megamek/client/ui/swing/MovementDisplay.java [hunk 2]
```diff
@@ -5233,64 +5238,52 @@ public synchronized void actionPerformed(ActionEvent ev) {
...
 
             // Loop through the players vector and fill in the arrays
             int idx = 0;
-            for (int i = 0; i < players.size(); i++) {
-                IPlayer p = players.get(i);
-                // If this is us, we skip it since we can't transfer to
-                // ourselves
-                if (p.getName().equals(
-                        clientgui.getClient().getLocalPlayer().getName())) {
+            for (var player: players) {
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/MovementDisplay.java:5249]:
```suggestion
            for (var player : players) {
```

File: megamek/src/megamek/client/ui/swing/MovementDisplay.java [hunk 3]
```diff
@@ -5233,64 +5238,52 @@ public synchronized void actionPerformed(ActionEvent ev) {
...
+                if (player.getName().equals(clientgui.getClient().getLocalPlayer().getName())
+                        || player.getTeam() == IPlayer.TEAM_UNASSIGNED) {
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/MovementDisplay.java:5251]: brackets

File: megamek/src/megamek/server/commands/TraitorCommand.java [hunk 4]
```diff
@@ -32,29 +49,21 @@ public void run(int connId, String[] args) {
...
+            } else if (player.getTeam() == IPlayer.TEAM_UNASSIGNED) {
+                server.sendServerChat(connId, "Player must be assigned a team.");
+            } else if (pid == connId) {
+                server.sendServerChat(connId, "You can't switch to the same side.");
+            } else {
                 server.sendServerChat(connId, ent.getDisplayName() + " will switch to " + player.getName() + "'s side at the end of this turn.");
                 ent.setTraitorId(pid);
             }
-            
         } catch (NumberFormatException nfe) {
```
Comment (Windchild292) [inline megamek/src/megamek/server/commands/TraitorCommand.java:66]:
```suggestion
        } catch (NumberFormatException ignored) {
```
Prevents an IDE warning on my end, plus is easier to read IMO.

**Review by Windchild292 — COMMENTED** [review]
Looks pretty good, just a few comments

File: megamek/src/megamek/client/ui/swing/MovementDisplay.java (hunk 2 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/MovementDisplay.java:5249]: Nah.

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**